rtmbot/core.py

```diff
@@ -183,23 +223,21 @@ def do_jobs(self):
     def do_output(self):
         output = []
         while True:
-            if 'outputs' in dir(self.module):
-                if len(self.module.outputs) > 0:
-                    logging.info("output from {}".format(self.module))
-                    output.append(self.module.outputs.pop(0))
-                else:
-                    break
+            if len(self.outputs) > 0:
+                logging.info("output from {}".format(self.name))
+                output.append(self.outputs.pop(0))
             else:
-                self.module.outputs = []
+                break
         return output
 
 
 class Job(object):
-    def __init__(self, interval, function, debug):
+    def __init__(self, interval, function, debug, plugin):
         self.function = function
```

Comment:
What do you think about simplifying the interface for `Job`? Instead of naming a function, can the interface be limited to just a single `run` or `output` function that all subclasses override?